compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp

```diff
@@ -691,12 +692,156 @@ FailureOr<Flow::WorkloadBuilder> getWorkloadBuilder(OpBuilder &builder,
   return result;
 }
 
+/// Finds contiguous loop ranges from the given linalg.generic.
+static SmallVector<ReassociationIndices> getContiguousLoops(
+    linalg::GenericOp genericOp, SmallVectorImpl<unsigned> &loopDims) {
+  SmallVector<ReassociationIndices> contiguousLoops;
+  if (loopDims.size() < 2) return contiguousLoops;
+
+  AffineMap firstMap = genericOp.getIndexingMapsArray().front();
+  // Checks all Affinemaps positions are the same.
+  auto isAllAffinemapsContiguous = [&](unsigned dim) {
+    if (firstMap.getNumResults() <= dim) return false;
+    return llvm::all_of(genericOp.getIndexingMapsArray(), [=](AffineMap map) {
+      if (map.getNumResults() <= dim) return false;
+      return map.getDimPosition(dim) == firstMap.getDimPosition(dim);
+    });
+  };
+
+  // Finds that d2-d3 and d6-d7 are contiguous for the case (d1, d0, d2, d3, d5,
+  // d4, d6, d7).
+  ReassociationIndices range;
+  for (auto [idx, dim] : llvm::enumerate(loopDims)) {
+    if (!range.empty()) {
+      if (!isAllAffinemapsContiguous(dim) || (range.back() + 1 != dim)) {
+        if (range.size() > 1)
+          contiguousLoops.push_back({range.begin(), range.end()});
+        range.clear();
+      }
+      if (!range.empty() && !isAllAffinemapsContiguous(dim)) range.clear();
```

Comment:
This is a redundant check. The range is already cleared when meat the condition.

If range is empty(), this is not needed. So let's discuss the case if it is not empty. If `!isAllAffinemapsContiguous(dim)` is true, the range will be cleared in the above condition.

In both cases, the check is redundant to me.